tofhir-engine/src/test/scala/io/tofhir/test/engine/execution/FileStreamInputArchiverTest.scala

```diff
@@ -0,0 +1,367 @@
+    val mockFileSystemSourceSettings = mock[FileSystemSourceSettings]
+    val mockFileSystemSource = mock[FileSystemSource]
+    val mockMappingTask = mock[FhirMappingTask]
+    val sourceFolderPath = "test-archiver-batch"
+    val inputFilePath = "test-input-file"
+    val jobId = "mocked_job_id_2"
+    when(mockTaskExecution.id).thenReturn(jobId)
+    when(mockTaskExecution.job).thenReturn(mockJob)
+    when(mockTaskExecution.mappingTasks).thenReturn(Seq(mockMappingTask))
+    when(mockJob.dataProcessingSettings).thenReturn(mockDataProcessingSettings)
+    when(mockDataProcessingSettings.archiveMode).thenReturn(ArchiveModes.DELETE)
+    when(mockJob.sourceSettings).thenReturn(Map(("_"->mockFileSystemSourceSettings)))
+    when(mockFileSystemSourceSettings.dataFolderPath).thenReturn(sourceFolderPath)
+    when(mockMappingTask.sourceContext).thenReturn(Map(("_" -> mockFileSystemSource)))
+    when(mockFileSystemSource.path).thenReturn(inputFilePath)
+
+    // Create a test input file
+    val inputFile = initializeInputFiles(sourceFolderPath, inputFilePath)
+
+    // Check whether input file exists
+    inputFile.exists() shouldBe true
+
+    //Call archiving function
+    FileStreamInputArchiver.applyArchivingOnBatchJob(mockTaskExecution)
+
+    // Check whether input file is deleted
+    inputFile.exists() shouldBe false
+
+    // Clean test directories
+    org.apache.commons.io.FileUtils.deleteDirectory(FileUtils.getPath(ToFhirConfig.engineConfig.contextPath, sourceFolderPath).toFile)
+  }
+
+  "FileStreamInputArchiver" should "get commit file" in {
+    // Create a mock job
+    val mockJob = mock[FhirMappingJob]
+    val mappingUrl = "mocked_mapping_url"
+    val jobId = "mocked_job_id"
+    when(mockJob.id).thenReturn(jobId)
+
+    // Create an execution
+    val testExecution = FhirMappingJobExecution(job = mockJob)
+
+    // Test whether commit directory is right
+    testExecution.getCommitDirectory(mappingUrl) shouldBe
+      FileUtils.getPath(ToFhirConfig.sparkCheckpointDirectory, jobId, mappingUrl.hashCode.toString, "commits").toString
+  }
+
+  "FileStreamInputArchiver" should "get source file" in {
+    // Create a mock job
+    val mockJob = mock[FhirMappingJob]
+    val mappingUrl = "mocked_mapping_url"
+    val jobId = "mocked_job_id"
+    when(mockJob.id).thenReturn(jobId)
+
+    // Create an execution
+    val testExecution = FhirMappingJobExecution(job = mockJob)
+
+    // Test whether source directory is right
+    testExecution.getSourceDirectory(mappingUrl) shouldBe
+      FileUtils.getPath(ToFhirConfig.sparkCheckpointDirectory, jobId, mappingUrl.hashCode.toString, "sources", "0").toString
+  }
+
+  "FileStreamInputArchiver" should "get input files" in {
+
+    val mappingUrl = "mocked_mapping_url"
+    val jobId = "mocked_job_id_3"
+
+    // Create a source file to refer location of test.csv
+    val sourceFile = FileUtils.getPath("test-archiver", jobId, mappingUrl.hashCode.toString, "sources", "0", "0").toFile
```

Comment:
Creating path with "sources", "0" is duplicated in many places. This logic should be implemented in a single place such as "SparkUtil". This would ease for instance handling a change in Spark's way of running.